web/src/containers/Main/hooks/index.ts

```diff
@@ -55,20 +133,47 @@ export function useMain(questionnaireId: string) {
         [questionnaireId, fhirMode],
     );
 
+    const [mapperInfoList, setMapperInfoList] = useState<MapperInfo[]>([]);
+    const [showModal, setShowModal] = useState(false);
+
+    useEffect(() => {
+        if (!showModal && mapperInfoList.length > 0) {
+            questionnaireManager.reload();
+            setMapperInfoList([]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why are you using react-hooks/exhaustive-deps here?
Can  `questionnaireManager.reload()` and `setMapperInfoList([])` be called in callback instead of effect.